src/discussions/posts/PostsView.jsx

```diff
@@ -1,17 +1,80 @@
+import { AppContext } from '@edx/frontend-platform/react';
+import { Spinner } from '@edx/paragon';
 import PropTypes from 'prop-types';
-import React from 'react';
-import Post, { postShape } from './post/Post';
+import React, { useContext, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router';
+import { RequestStatus } from '../../data/constants';
+import {
+  selectAllThreads,
+  selectThreadAllPostsTypeFilter,
+  selectThreadSorting,
+  selectThreadStatusFilter,
+  selectTopicThreads,
+  selectUserThreads,
+  threadsLoadingStatus,
+} from './data/selectors';
+import { fetchThreads } from './data/thunks';
+import PostFilterBar from './post-filter-bar/PostFilterBar';
+import PostLink from './post/PostLink';
+
+function PostsView({ mine }) {
+  const {
+    courseId,
+    topicId,
+  } = useParams();
+  const dispatch = useDispatch();
+
+  const { authenticatedUser } = useContext(AppContext);
+  const orderBy = useSelector(selectThreadSorting());
+  const statusFilter = useSelector(selectThreadStatusFilter());
+  const typeFilter = useSelector(selectThreadAllPostsTypeFilter());
+  const loadingStatus = useSelector(threadsLoadingStatus());
+
+  let posts = [];
+  if (topicId) {
+    posts = useSelector(selectTopicThreads(topicId));
+  } else if (mine) {
+    posts = useSelector(selectUserThreads(authenticatedUser.username));
+  } else {
+    posts = useSelector(selectAllThreads());
+  }
```

Comment:
This is a pretty weird construction but I couldn't find where the component is used; if it's easy to do, I'd pull this up to the parent.

Reply:
This is the topmost view.